gemhardware/include/gem/hw/HwGenericAMC.h

```diff
@@ -84,38 +69,48 @@ namespace gem {
         /** GEM system information **/
         /****************************/
         /**
-         * Read the board ID registers
+         * @brief Read the board ID register (should be reimplemented in derived HW, e.g., GLIB)
+         * @param legacy determines whether to read the legacy system board ID (GLIB)
+         *        or the GEM_AMC board ID
          * @returns the AMC board ID as a std::string
          */
-        virtual std::string getBoardID();
+        virtual std::string getBoardID(bool const legacy=false);
```

Comment:
Use a pure virtual function to document the need for implementation in derived classes?
I couldn't read the whole PR yet, but I assume there is a hierarchy like:
```
HWGenericAMC
    HWAMC
    HWGLIB
```
I'll let @lpetre-ulb comment, but I don't think there will be any visible difference between the AMC and the GLIB other than the hostname.
Also, `const`?

Reply:
This hierarchy is historical, but probably before `v2.0.0` all AMC specialization will be removed.
In the past the `GLIB` had a user and system firmware block, and `GEM_AMC` basically was providing the user block, with no promise of continued support of the system block (SRAMs, hardware specific features, etc.), so the `GenericAMC` part corresponds to what `GEM_AMC` provides generically, across all (potentially) supported hardware, and in this specific case, both blocks provided a "board ID"...
I envision having a dedicated discussion on exactly this topic in the semi-near future.

Yes, all getters *should* be `const`, but there are some member variable modifications that *can* happen the way things are currently implemented (re `IPBusErrors` mentioned above) that need to be cleaned up for a more "correct" interface.